### PR TITLE
fix ignore settings

### PR DESCRIPTION
**Pull request by Nik-Sch**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -176,7 +176,7 @@
 							"type": "string"
 						},
 						"default": [],
-						"description": "Glob expressions for files to be ignored by the project parser"
+						"description": "Glob expressions for files to be ignored by the project parser (should match the file and not folder! i.e. **/generated/**/* instead of **/generated/*)"
```
Comment (g0t00) [inline package.json:179]: So the glob has to match the whole path of the file, and does not hit when the parent folder is ignored?
I do not think this is optimal. i would prefer if this works as in gitgnore.
i. if I ignor a folder all children will be automatically ignored

File: test/unitTests/ignoreFiles/ignoreFiles.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return await ProjectParser.create([pathToFileURL(__dirname)]);
+}
+
+test.each([
+  ["**/*entity*", true], // by filename
+  ["**/*no_match*", false], // by filename
+  ["**/export/**/*", true], // by last folder
+  ["**/modules/**/*", true], // by middle folder
+  ["src/*", false], // by first folder
+  ["src/**/*", true], // by first folder
```
Comment (g0t00) [inline test/unitTests/ignoreFiles/ignoreFiles.test.ts:21]: Yeah again , i do not like this behavior. I do not think this is clear from user perspective.
The first think ignores the src/modules folder I think this should then everythink inside.

File: test/unitTests/ignoreFiles/ignoreFiles.test.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+])('Test ignore glob "%s". errorExpected: %s', async (ignoreGlob: string, errorExpected: boolean) => {
```
Comment (g0t00) [inline test/unitTests/ignoreFiles/ignoreFiles.test.ts:22]: Add testcase without glob, eg.
`src`, `src/`, `does-not-exist`, `does-not-exist/src`, `does-not-exist/**/src`, , `does-not-exist/**`

File: package.json (hunk 1 above)
Comment (Nik-Sch) [reply package.json:179]: https://git-scm.com/docs/gitignore#_pattern_format
I believe exactly emulating the gitignore pattern would require us to implement our own glob matching...
But, yes we could also apply it to folders. I will have a look at this

**Review by g0t00 — APPROVED** [review]

**Pull request merged**